mypy_django_plugin/transformers/querysets.py

```diff
@@ -168,7 +178,13 @@ def extract_proper_type_queryset_values_list(ctx: MethodContext, django_context:
     row_type = get_values_list_row_type(
         ctx, django_context, django_model.cls, is_annotated=django_model.is_annotated, flat=flat, named=named
     )
-    return default_return_type.copy_modified(args=[django_model.typ, row_type])
+    ret = default_return_type.copy_modified(args=[django_model.typ, row_type])
+    if not named and (field_lookups := resolve_field_lookups(ctx.args[0], django_context)):
+        # For non-named values_list, the row type does not encode column names.
+        # Attach selected field names to the returned QuerySet instance so that
+        # subsequent annotate() can make an informed decision about name conflicts.
+        ret.extra_attrs = ExtraAttrs(attrs={}, immutable=set(field_lookups), mod_name=None)
```

Comment:
Question: shouldn't we also set `attrs` when setting `immutable`?

Reply:
From what I understand of this `ExtraAttrs` thingy, it's not supposed to be used like that and we are kinda hacking around to add metadata per `Instance`. I can add the attrs dict but we will ignore it and I think mypy too :/

Reply:
Btw, consider using `copy_with_extra_attr`?

Reply:
I would still prefer to preserve, some other plugin / feature might modify it.